### PR TITLE
PWGHF: implement check on prompt/non prompt toXiPi tasks

### PR DESCRIPTION
**Pull request by ZFederica**
- implement check on charm baryon origin in toXiPi creator
- add the same information in the treeCreator table

### CODE REVIEW

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx [hunk 1]
```diff
@@ -444,7 +445,7 @@ struct HfCandidateCreatorToXiPiMc {
     // Match reconstructed candidates.
     for (const auto& candidate : candidates) {
       flag = 0;
-      // origin = 0;
+      origin = 0;
```
Comment (fgrosa) [inline PWGHF/TableProducer/candidateCreatorToXiPi.cxx:448]:
```suggestion
      origin = RecoDecay::OriginType::None;
```

File: PWGHF/TableProducer/candidateCreatorToXiPi.cxx [hunk 2]
```diff
@@ -505,23 +514,29 @@ struct HfCandidateCreatorToXiPiMc {
...
     // Match generated particles.
     for (const auto& particle : mcParticles) {
-      flag = -9;
+      flag = 0;
       sign = -9;
       debugGenCharmBar = 0;
       debugGenXi = 0;
       debugGenLambda = 0;
-      // origin = 0;
+      origin = 0;
```
Comment (fgrosa) [inline PWGHF/TableProducer/candidateCreatorToXiPi.cxx:539]:
```suggestion
      origin = RecoDecay::OriginType::None;
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 3]
```diff
@@ -284,7 +286,7 @@ struct HfTreeCreatorToXiPi {
     // Filling candidate properties
     rowCandidateFull.reserve(candidates.size());
     for (const auto& candidate : candidates) {
-      fillCandidate(candidate, -7, -7);
+      fillCandidate(candidate, -7, -7, -7);
```
Comment (fgrosa) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:289]: Why `-7`? Shouldn't it be better to fill it with `RecoDecay::OriginType::None`?

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @ZFederica I just have a couple of minor comments, then we can merge it!

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 3 above)
Comment (ZFederica) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx]: Hi @fgrosa, thanks for the comments! Sure, I change it

**Pull request merged**